lib/expression/callbacks/eval_helpers.ex

```diff
@@ -8,7 +8,10 @@ defmodule Expression.Callbacks.EvalHelpers do
   @spec eval!(term, map) :: term
   def eval!(ast, ctx) do
     ast
-    |> Expression.Eval.eval!(ctx)
+    |> Expression.Eval.eval!(
+      ctx,
+      Map.get(ctx, "__CUSTOM_CALLBACK__", Expression.Callbacks.Standard)
```

Comment:
This feels like we need to give this some more thought, it's an entirely different means of specifying the callback module. In other places we're passing it in explicitly as an optional argument.

Maybe this helper just isn't useful enough to be defined in a separate module, would that prevent us having to supply the callback module as an argument? Especially because the eval! function is used in the callback module primarily.

Reply:
I assumed that this was done before the implantation of “or” in the same way. I think it is fixed because essentially this code was duplicating what “or” is doing in the body of its function 🤔And yes I agree that definining the callback like that changes things quite a bit but I haven’t found a better way to pass it as argument to all functions without changing all of their arity (and so make it also incompatible with all existing callback modules), any idea of how to still solve the issue with the test and keep it as argument?